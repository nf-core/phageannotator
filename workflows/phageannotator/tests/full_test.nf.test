nextflow_workflow {

    name "Test workflow: PHAGEANNOTATOR"
    script "workflows/phageannotator/main.nf"
    workflow "PHAGEANNOTATOR"

    // Dependencies
    tag "SEQKIT_SEQ"
    //tag"APPENDSCREENHITS"
    tag "QUALITYFILTERVIRUSES"
    tag "ANICLUSTER_ANICALC"
    tag "ANICLUSTER_ANICLUST"
    tag "ANICLUSTER_EXTRACTREPS"
    tag "COVERM_CONTIG"
    tag "INSTRAIN_STB"
    tag "FASTQ_VIRUS_ENRICHMENT_VIROMEQC"
    tag "FASTQ_FASTA_REFERENCE_CONTAINMENT_MASH"
    tag "FASTA_VIRUS_CLASSIFICATION_GENOMAD"
    tag "FASTA_VIRUS_QUALITY_CHECKV"
    tag "FASTA_ALL_V_ALL_BLAST"
    tag "FASTA_PHAGE_HOST_IPHOP"
    tag "FASTA_PHAGE_FUNCTION_PHAROKKA"
    tag "FASTA_MICRODIVERSITY_INSTRAIN"
    tag "CAT_CAT"
    tag "BOWTIE2_BUILD"
    tag "GENOMAD_ENDTOEND"
    tag "GUNZIP"
    tag "BACPHLIP"
    tag "FASTQ_ALIGN_BOWTIE2"


    test("Parameters: Run all tools") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ],
                    [
                        [ id:'test2' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ],
                    [
                        [ id:'test2' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ]
                )
                """
            }
            params {
                outdir                      = "$outputDir"
                run_bacphlip                = true
                run_genomad_taxonomy        = true
                run_instrain                = true
                run_iphop                   = true
                run_reference_containment   = true
                run_viromeqc                = true
                // parameters to decrease sensitivity for test data
                mash_screen_min_score   = 0.01
                genomad_min_score       = 0.01
                genomad_max_fdr         = 1
                // speed up options since tools are fully tested in subworkflows
                genomad_disable_nn          = true
                genomad_sensitivity         = 0.1
                checkv_minimal_db           = true
                iphop_partial_test          = true
                // lower instrain sensitivity so tool runs on test data
                instrain_min_ani            = 0.01
                instrain_min_mapq           = 2
                instrain_min_variant_cov    = 5
                instrain_min_snp_freq       = 0.01
                instrain_max_snp_fdr        = 1
                instrain_min_genome_cov     = 0.01
                instrain_popani_thresh      = 0.01
                instrain_min_genome_comp    = 0.01
                instrain_min_genome_breadth = 0.01
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert !path("${outputDir}/VirusLifestyle/bacphlip/test.filtered.fna.bacphlip").exists() },
                { assert path("${outputDir}/VirusAbundance/coverm/contig/all_samples_alignment_results.tsv").readLines().size() == 4 }
            )
        }
    }
}
