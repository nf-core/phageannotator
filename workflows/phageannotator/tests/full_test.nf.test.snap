{
    "Parameters: Run all tools": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.viromeqc.tsv:md5,728ab79cbb6c7afd0dc3f38c5ddfe642"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,b55f2c77b63adc41da980568819d1aae"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.fasta_w_screen_hits_virus_summary.tsv:md5,617215f2a883d4ec8d95aacd0b09d8a1"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits_virus_summary.tsv:md5,f5b59da727b9793461f8b85cf7c818df"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test2"
                        },
                        "quality_summary.tsv:md5,db5422a546af97a3d997045b3eb9f321"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "quality_summary.tsv:md5,f5273c5640e76d64d46c54c09f75f2d9"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,741ab7b127375e64069553fa51b2eac9"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,a3f4a97cd394993502c30dd444bc1b88"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,07bf1a1918d0a5aa2422c7b1a3b1c085"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "Host_prediction_to_genus_m90.csv:md5,528dcefb6a6c743b43f0dbc40bec3692"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives_taxonomy.tsv:md5,fb8440b1fd32f96414b71b76ec228df1"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_cds_final_merged_output.tsv:md5,26c4d18972b491b05e844cff84fc63ad"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.IS_gene_info.tsv:md5,d7dd99000ba61cc6c2032183ab75ddcd"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.IS_gene_info.tsv:md5,d7dd99000ba61cc6c2032183ab75ddcd"
                    ]
                ],
                "9": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,1097ac7d62d13e9e74916d9dc1fabb7b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,2b068b9678f63d6bb06a2a3652e92ba9",
                    "versions.yml:md5,2d1864a0523e2738f150eaf87d4472d5",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,3c8b8ac5a86d6d7fba86fc61f4d5fca9",
                    "versions.yml:md5,42ce3c1803e6f3030b60d1d4610b629d",
                    "versions.yml:md5,478c8fdcebf1357f40c26d7991d85eb4",
                    "versions.yml:md5,5074b29ede738ba6d39ef056e1cde929",
                    "versions.yml:md5,5bf636608598d8b495d8b38fcee37dd0",
                    "versions.yml:md5,6323b255a9c1b9ed3e1a41aea5bfa2bb",
                    "versions.yml:md5,716410d7e9dc048a64ada71202cbc23e",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,afc743a69ab04322fbc732b24e18d208",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,c0598196e761dbcdc90ff95a2f623b9c",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,d73a3ecea014f00c8d95513555c42d23",
                    "versions.yml:md5,dc43c8404f9f95b46c1a9c8d51687032",
                    "versions.yml:md5,dc43c8404f9f95b46c1a9c8d51687032",
                    "versions.yml:md5,de688faac0a9124c4f2db74ce1b2c795",
                    "versions.yml:md5,dfa7110695181c4e8daa47d384563588"
                ],
                "anicluster_reps_fna_gz": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,07bf1a1918d0a5aa2422c7b1a3b1c085"
                    ]
                ],
                "filtered_viruses_fna_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,741ab7b127375e64069553fa51b2eac9"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,a3f4a97cd394993502c30dd444bc1b88"
                    ]
                ],
                "host_predictions_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "Host_prediction_to_genus_m90.csv:md5,528dcefb6a6c743b43f0dbc40bec3692"
                    ]
                ],
                "instrain_gene_info": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.IS_gene_info.tsv:md5,d7dd99000ba61cc6c2032183ab75ddcd"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.IS_gene_info.tsv:md5,d7dd99000ba61cc6c2032183ab75ddcd"
                    ]
                ],
                "marker_taxonomy_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives_taxonomy.tsv:md5,fb8440b1fd32f96414b71b76ec228df1"
                    ]
                ],
                "pharokka_output_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_cds_final_merged_output.tsv:md5,26c4d18972b491b05e844cff84fc63ad"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,1097ac7d62d13e9e74916d9dc1fabb7b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,2b068b9678f63d6bb06a2a3652e92ba9",
                    "versions.yml:md5,2d1864a0523e2738f150eaf87d4472d5",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,3c8b8ac5a86d6d7fba86fc61f4d5fca9",
                    "versions.yml:md5,42ce3c1803e6f3030b60d1d4610b629d",
                    "versions.yml:md5,478c8fdcebf1357f40c26d7991d85eb4",
                    "versions.yml:md5,5074b29ede738ba6d39ef056e1cde929",
                    "versions.yml:md5,5bf636608598d8b495d8b38fcee37dd0",
                    "versions.yml:md5,6323b255a9c1b9ed3e1a41aea5bfa2bb",
                    "versions.yml:md5,716410d7e9dc048a64ada71202cbc23e",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,afc743a69ab04322fbc732b24e18d208",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,c0598196e761dbcdc90ff95a2f623b9c",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,d73a3ecea014f00c8d95513555c42d23",
                    "versions.yml:md5,dc43c8404f9f95b46c1a9c8d51687032",
                    "versions.yml:md5,dc43c8404f9f95b46c1a9c8d51687032",
                    "versions.yml:md5,de688faac0a9124c4f2db74ce1b2c795",
                    "versions.yml:md5,dfa7110695181c4e8daa47d384563588"
                ],
                "virus_classification_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.fasta_w_screen_hits_virus_summary.tsv:md5,617215f2a883d4ec8d95aacd0b09d8a1"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits_virus_summary.tsv:md5,f5b59da727b9793461f8b85cf7c818df"
                    ]
                ],
                "virus_enrichment_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.viromeqc.tsv:md5,728ab79cbb6c7afd0dc3f38c5ddfe642"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,b55f2c77b63adc41da980568819d1aae"
                    ]
                ],
                "virus_quality_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "quality_summary.tsv:md5,db5422a546af97a3d997045b3eb9f321"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "quality_summary.tsv:md5,f5273c5640e76d64d46c54c09f75f2d9"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-16T23:12:59.497752604"
    }
}